src/BeadingStrategy/CenterDeviationBeadingStrategy.cpp

```diff
@@ -11,58 +11,79 @@ CenterDeviationBeadingStrategy::Beading CenterDeviationBeadingStrategy::compute(
     Beading ret;
 
     ret.total_thickness = thickness;
-    if (bead_count > 0)
+    if (bead_count == 1)
     {
-        for (coord_t bead_idx = 0; bead_idx < bead_count / 2; bead_idx++)
+        ret.bead_widths.emplace_back(thickness);
+        ret.toolpath_locations.emplace_back(thickness / 2);
+    }
+    else if (bead_count > 1)
+    {
+        // first half minus middle:
+        ret.bead_widths.emplace_back(optimal_width_outer);
+        ret.toolpath_locations.emplace_back(optimal_width_outer / 2);
+        for (coord_t bead_idx = 1; bead_idx < bead_count / 2; bead_idx++)
         {
-            ret.bead_widths.emplace_back(optimal_width);
-            ret.toolpath_locations.emplace_back(optimal_width * (bead_idx * 2 + 1) / 2);
+            ret.bead_widths.emplace_back(optimal_width_inner);
+            ret.toolpath_locations.emplace_back(optimal_width_outer + optimal_width_inner * ((bead_idx - 1) * 2 + 1) / 2);
         }
+
+        // middle (if any):
+        const coord_t optimal_width_middle = bead_count == 1 ? optimal_width_outer : optimal_width_inner;
```

Comment:
In this case, the bead_count will always be > 1, so the optimal_width_middle is always the optimal_width_inner.

Reply:
Ah yes, made the exception for 1 bead count after I made this. It can be removed in that case.